jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleResolver.java

```diff
@@ -369,16 +369,34 @@ private static final MethodHandle sendResolveMethodHandle(
 			String name,
 			String typeDescriptor,
 			ClassLoader loader) throws Throwable {
-		try {
+		MethodType type = null;
 /*[IF OPENJDK_METHODHANDLES]*/
+		switch (cpRefKind) {
+		case 1: /* getField */
+		case 2: /* getStatic */
+		case 3: /* putField */
+		case 4: /* putStatic */
+			type = MethodTypeHelper.vmResolveFromMethodDescriptorString("(" + typeDescriptor + ")V", loader, null);
+			break;
+		case 5: /* invokeVirtual */
+		case 6: /* invokeStatic */
+		case 7: /* invokeSpecial */
+		case 8: /* newInvokeSpecial */
+		case 9: /* invokeInterface */
+			type = MethodTypeHelper.vmResolveFromMethodDescriptorString(typeDescriptor, loader, null);
+			break;
+		default:
+			/* Can never happen */
+			throw new UnsupportedOperationException();
```

Comment:
Throw an InternalError here with details of the problem

Reply:
Completed.